charts/astronomer/templates/houston/api/houston-deployment.yaml

```diff
@@ -33,6 +33,16 @@ spec:
       annotations:
         checksum/houston-config: {{ include (print $.Template.BasePath "/houston/houston-configmap.yaml") . | sha256sum }}
         checksum/houston-backend-secret: {{ include (print $.Template.BasePath "/houston/api/houston-backend-secret.yaml") . | sha256sum }}
+        {{- if and .Values.global.nats.jetStream.enabled .Values.global.nats.jetStream.tls }}
+        {{- $jetStreamTLSSecretName := printf "%s-client" (include "nats.jestreamTLSSecret" . ) }}
+        {{- $secretObj := (lookup "v1" "Secret" .Release.Namespace $jetStreamTLSSecretName) | default dict }}
+        {{- $secretData := (get $secretObj "data") | default dict }}
+        {{- if $secretData }}
+        checksum/jetstream-tls-cert: {{ sha256sum (b64dec (get $secretData "tls.crt")) }}
+        {{- else }}
+        checksum/jetstream-tls-cert: {{ printf "template-checksum-%s-%t-%t" $jetStreamTLSSecretName | sha256sum }}
```

Comment:
why do we need an else logic here

Reply:
you can use --client as server to render and if this is only for passing tests we can mock it i feel else might not be even needed you can verify if the key exists in annotation not the contents